Script/AtomicNET/AtomicNET/Scene/CSComponentCore.cs

```diff
@@ -332,6 +332,21 @@ void HandleComponentLoad(uint eventType, ScriptVariantMap eventData)
 
         }
 
+        void HandleComponentAssemblyReference(uint eventType, ScriptVariantMap eventData)
+        {
+#if ATOMIC_DESKTOP || ATOMIC_ANDROID
+            string assemblyPath = eventData["AssemblyPath"];
```

Comment:
I am pretty sure we're going to need to make design/edit time systems for this... at runtime, we should ensure all the types are loaded, ahead of being referenced.  Probably not the last time this code will be touched :)